### PR TITLE
[patch] Documentation: add concept section to the architecture document

### PR DESCRIPTION
**Pull request by kevindiu**
### Description:

This document add a concept section to the architecture document to give an overall concept of Vald before deep diving into details.

### Environment:

- Golang Version: 1.14.3
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.11.5

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [x] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/overview/architecture.md [hunk 1]
```diff
@@ -15,17 +16,61 @@ Vald uses a cloud-native architecture focusing on [Kubernetes](https://kubernete
...
+
+  Vald is designed based on the microservice architecture. Vald components are highly decoupled into small components and connected, which increases the overall agility and maintainability of Vald.
+
+- Containerized
+
+  All components in Vald are containerized, which means you can easily deploy Vald components on any environment.
+
+- Observability & Real-time monitoring
+
+  Vald support pprof and Prometheus to monitor and trace the log in every Vald components.
```
Comment (hlts2) [inline docs/overview/architecture.md:33]:
```suggestion
  Vald supports pprof and Prometheus to monitor and trace the log in every Vald components.
```

**Comment by vankichi** [comment]
@kpango @rinx @kmrmt 
We (SET team) considered we should describe about Vald's basic concepts (e.g. cloud native, apply micro service architecture, not deep dive each component) before showing Vald's overall architecture images / data flows to users because it will help for new users.
So, we added Vald's basic concepts section at the top of architecture document.
We'd like to get authors' review whether added document is correct or not.
Could you please review by 6/9? And, we appreciate you if you can review ASAP.
thank you for your cooperation 🙏

**Comment by kpango** [comment]
@vankichi thank you for your team's great job, this week I'm so busy could you wait until 6/10?

**Comment by vankichi** [comment]
@kpango 
Sure ! But please remain that we'd like to publish this update by 6/12 (Fri.) because of KubeFest Tokyo 2020.

File: docs/overview/architecture.md [hunk 2]
```diff
@@ -15,17 +16,61 @@ Vald uses a cloud-native architecture focusing on [Kubernetes](https://kubernete
...
+
+  Vald supports pprof and Prometheus to monitor and trace the log in every Vald components.
+
+- Distributed
+
+  All the data and index are distributed to Vald Agents in the Vald cluster so whenever you search a vector, all Vald agents can process parallelly and merge the result by Vald LB Gateway.
+
+- Kubernetes based
+
+  Vald can integrate with kubernetes which enables the following features.
```
Comment (rinx) [inline docs/overview/architecture.md:41]: badge
```suggestion
  Vald can integrate with Kubernetes which enables the following features.
```

File: docs/overview/architecture.md [hunk 3]
```diff
@@ -15,17 +16,61 @@ Vald uses a cloud-native architecture focusing on [Kubernetes](https://kubernete
...
+
+  - Orchestrated
 
-We will explain this image in the following section.
+    Kubernetes supports container orchestration. All components in Vald can be managed by kubernetes automatically.
```
Comment (rinx) [inline docs/overview/architecture.md:45]: badge
```suggestion
    Kubernetes supports container orchestration. All components in Vald can be managed by Kubernetes automatically.
```

File: docs/overview/architecture.md [hunk 4]
```diff
@@ -15,17 +16,61 @@ Vald uses a cloud-native architecture focusing on [Kubernetes](https://kubernete
...
 
+  - Horizontal scalable
+
+    All Vald components are designed and implemented to be scalable. You can add any node in kubernetes cluster at any time to scale your kuberentes cluster, or changing the number of replicas to scale Vald.
```
Comment (rinx) [inline docs/overview/architecture.md:49]: badge
```suggestion
    All Vald components are designed and implemented to be scalable. You can add any node in Kubernetes cluster at any time to scale your Kuberentes cluster, or changing the number of replicas to scale Vald.
```

File: docs/overview/architecture.md [hunk 5]
```diff
@@ -15,17 +16,61 @@ Vald uses a cloud-native architecture focusing on [Kubernetes](https://kubernete
...
+
+    Kubernetes supports the auto-healing feature. The pod can start a new instance automatically whenever the pod is down.
+
+  - Data persistency
+
+    Vald implements backup features. Whenever a Vald Agent is down and kubernetes start a new Vald Agent instance, the data is automatically restored to the new instance to prevent data loss.
+
+  - Easy to manage
+
+    We implement helm chart deployment to easily deploy Vald on your kubernetes cluster and we implemented custom resources to manage and operate with Vald.
```
Comment (rinx) [inline docs/overview/architecture.md:61]: badge
```suggestion
    Vald can be deployed easily on your Kubernetes cluster by using Helm charts. The custom resources and custom controllers are useful to manage your Vald cluster.
```

File: docs/overview/architecture.md [hunk 6]
```diff
@@ -15,17 +16,61 @@ Vald uses a cloud-native architecture focusing on [Kubernetes](https://kubernete
...
+
+  All components in Vald are containerized, which means you can easily deploy Vald components on any environment.
+
+- Observability & Real-time monitoring
+
+  Vald supports pprof and Prometheus to monitor and trace the log in every Vald components.
+
+- Distributed
+
+  All the data and index are distributed to Vald Agents in the Vald cluster so whenever you search a vector, all Vald agents can process parallelly and merge the result by Vald LB Gateway.
```
Comment (rinx) [inline docs/overview/architecture.md:37]: badge
agents don't know "all the data". they know just UUIDs and indices.

I think "distributed" is not so important, but "distributed vector spaces" is an important aspect of the architecture.

File: docs/overview/architecture.md [hunk 7]
```diff
@@ -15,17 +16,61 @@ Vald uses a cloud-native architecture focusing on [Kubernetes](https://kubernete
...
+
+  Vald is designed based on the microservice architecture. Vald components are highly decoupled into small components and connected, which increases the overall agility and maintainability of Vald.
+
+- Containerized
+
+  All components in Vald are containerized, which means you can easily deploy Vald components on any environment.
+
+- Observability & Real-time monitoring
+
+  Vald supports pprof and Prometheus to monitor and trace the log in every Vald components.
```
Comment (rinx) [inline docs/overview/architecture.md:33]: badge
Not only pprof and Prometheus exporter are the observability features. Please revise it like following:
```suggestion
  Every Vald components supports Cloud-Native based observability features, e.g. Prometheus exporter and Jaeger exporter.
```

File: docs/overview/architecture.md [hunk 8]
```diff
@@ -15,17 +16,61 @@ Vald uses a cloud-native architecture focusing on [Kubernetes](https://kubernete
...
+
+- Distributed
```
Comment (kevindiu) [inline docs/overview/architecture.md:35]:
```suggestion
- Distributed vector spaces
```

File: docs/overview/architecture.md (hunk 7 above)
Comment (kevindiu) [reply docs/overview/architecture.md:33]: thank you for your comment. I think it is better 🤔
```suggestion
  All Vald components support Cloud-Native based observability features such as Prometheus and Jaeger.
```

File: docs/overview/architecture.md (hunk 6 above)
Comment (kevindiu) [reply docs/overview/architecture.md:37]: Yes you are correct. What do you think the following changes?
```suggestion
  All the vector data and indexes are distributed to Vald Agents in the Vald cluster. Whenever you search a vector in Vald cluster, all Vald agents can process parallelly and merge the result by Vald LB Gateway.
```

File: docs/overview/architecture.md (hunk 7 above)
Comment (rinx) [reply docs/overview/architecture.md:33]: sounds good. :+1:
I prefer "Prometheus and Jaeger **exporter**".

Comment (kevindiu) [reply docs/overview/architecture.md:33]:
```suggestion
  All Vald components support Cloud-Native based observability features such as Prometheus and Jaeger exporter.
```

**Pull request merged**